src/user-guide/writing-workflows/runtime.rst

```diff
@@ -317,6 +319,26 @@ output files are typically held in cycle point sub-directories of this.
 The top level share directory location can be changed, e.g. to a large data
 area, by global config settings under :cylc:conf:`global.cylc[install][symlink dirs]`.
 
+If your workflow creates or installs executables or python libraries
+as it is running, these can be placed in:
+
+* ``share/bin`` - for executables.
+* ``share/lib/python`` - for Python modules.
+
+These folders will be added to ``PATH`` and ``PYTHONPATH`` so they can be
+used by jobs in the same way as
+:ref:`top level "bin/" and "lib/python/" directories <WorkflowDefinitionDirectories>`.
+
+.. note::
+
+   Cylc will not create these folders.
+
+.. important::
+
+   Scripts placed in these directories will be called instead of a script
+   of the same name in ``bin`` or ``lib/python`` directories at the top
+   level of the workflow.
```

Comment:
I would just state the facts here, in case users over-interpret that statement. Something like this:

```suggestion
  These locations will be placed first in the executable and Python module
  search paths, before those in the parent run directory .
```